core/src/actions_normalize.cpp

```diff
@@ -17,40 +15,13 @@
 #include "state.hpp"
 #include "option.hpp"
 #include "debuglog.h"
-#include "core_icu.h"
+#include "util_normalize.hpp"
+#include "utfcodec.hpp"
+#include "kmx/kmx_xstring.h"
 
-// forward declarations
-
-icu::UnicodeString context_items_to_unicode_string(km::core::context const *context);
-
-
-// TEMP
-namespace km {
-namespace core {
-namespace util {
+// forward declaration
+bool context_items_to_unicode_string(km::core::context const *context, std::u32string &str);
 
-/**
- * Helper to convert icu::UnicodeString to a UTF-32 km_core_usv buffer,
- * nul-terminated
- */
-inline km_core_usv *unicode_string_to_usv(icu::UnicodeString& src) {
```

Comment:
If I'm tracking the changes correctly, this removal is fine due to no longer using `icu::UnicodeString&` as the standard intermediary string type.  

Replacing it with `std::u32string` here and elsewhere... appears to avoid the need for all the ICU-related assertions, etc that previously existed here?